packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -2,6 +2,7 @@ import { IRole, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from '
 import * as lambda from '@aws-cdk/aws-lambda';
 import { Duration, IResource, Lazy, Resource, Stack, Token } from '@aws-cdk/core';
 import { Construct } from 'constructs';
+import { toASCII as punycodeEncode } from 'punycode/';
```

Comment:
```suggestion
import { toASCII as punycodeEncode } from 'punycode';
```

Reply:
Actually this notation is a must.
Punycode was originally a package which shipped with NodeJS and could be imported via `require('punycode')`. Although deprecated since v7, it is still part of NodeJS and importing from `punycode` without trailing `/` will import the deprecated package.

Reply:
I see. Can we write some kind of test or assertion that someone doesn't accidentally change this?

Reply:
Good point.
On Node v12, `require('punycode')` also loads the current `punycode`-package v2.1.0, although deprecated.
On a newer Node version (possibly v16+) it might just throw an error. Still an assertion is the best way.

A unit test I can think of is to read the installed version from `package.json`, patch `require` for this test, load `user-pool.ts`, check if `require` is invoked with `punycode/` as argument and restore the require function. Seems a bit hacky though.

Reply:
Are there any specific APIs or functionality that are in `punycode/` but not in `punycode` that can be used to assert this?

Reply:
The other option is to use an eslint rule like this one - https://eslint.org/docs/rules/no-restricted-modules

Reply:
E. g. in Node v12 `require('punycode')` returns the punycode package v2.1.0.
The added package `punycode` in this codebase is the same package v2.1.0.

So in Node v12-v15 it is identical, but it is marked as deprecated.

The eslint approach is better :+1: I'll add a rule to the PR.